packages/devtools_app/lib/src/shared/ui/filter.dart

```diff
@@ -463,3 +465,167 @@ extension PatternListExtension on List<Pattern> {
         : cast<String>();
   }
 }
+
+// TODO:: Change screens that use [DevtoolsFilterButton] to use a [StandaloneFilterField]
+// instead.
+/// A text field for controlling the filter query for a [FilterControllerMixin].
+///
+/// This text field has a button to open a dialog for toggling any toggleable
+/// text filters.
+class StandaloneFilterField<T> extends StatefulWidget {
+  const StandaloneFilterField({
+    super.key,
+    required this.controller,
+  });
+
+  final FilterControllerMixin<T> controller;
+
+  @override
+  State<StandaloneFilterField<T>> createState() =>
+      _StandaloneFilterFieldState<T>();
+}
+
+class _StandaloneFilterFieldState<T> extends State<StandaloneFilterField<T>>
+    with AutoDisposeMixin {
+  late final TextEditingController queryTextFieldController;
+  late bool useRegExp;
+
+  @override
+  void initState() {
+    super.initState();
+    queryTextFieldController = TextEditingController(
+      text: widget.controller.activeFilter.value.queryFilter.query,
+    );
+    useRegExp = widget.controller.useRegExp.value;
+    addAutoDisposeListener(widget.controller.useRegExp, () {
```

Comment:
Is `FilterControllerMixin<T>.useRegExp` a `ValueListenable`? If so, we should just use a `ValueListenableBuilder` in the build method and not register a listener here.